src/main/java/main/controllers/Project/SuiteController.java

```diff
@@ -86,4 +90,60 @@ public boolean delete(TestSuiteDto template) throws AqualityException {
         }
         return testSuites;
     }
+
+    public void syncLegacyTests(List<TestDto> legacyTests, Integer suiteId, boolean removeNotExecutedResults) throws AqualityException {
+        if(baseUser.isManager() || baseUser.getProjectUserBySuiteId(suiteId).isManager()){
+            legacyTests.forEach(test -> {
+                Test2SuiteDto test2Suite = new Test2SuiteDto();
+                test2Suite.setTest_id(test.getId());
+                test2Suite.setSuite_id(suiteId);
+                try {
+                    test2SuiteDao.delete(test2Suite);
+                    if(removeNotExecutedResults) {
+                        removePendingResultsForTest(suiteId, test.getId());
+                    }
+                } catch (AqualityException e) {
+                    e.printStackTrace();
+                }
+            });
+        }else{
+            throw new AqualityPermissionsException("Account is not allowed to Sync Test Suite", baseUser);
+        }
+    }
+
+    private void removePendingResultsForTest(Integer suiteId, Integer testId) throws AqualityException {
+        List<TestResultDto> testResults = testResultDao.selectSuiteLegacyResults(suiteId, testId);
+        testResults.forEach(testResult -> {
+            try {
+                testResultDao.delete(testResult);
+            } catch (AqualityException e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    public List<TestDto> findLegacyTests(Integer suiteId, Integer notExecutedFor) throws AqualityException {
+        if (baseUser.isManager() || baseUser.getProjectUserBySuiteId(suiteId).isManager()) {
+            TestRunDto testRunTemplate = new TestRunDto();
+            testRunTemplate.setTest_suite_id(suiteId);
+            testRunTemplate.setLimit(notExecutedFor);
+            List<TestRunDto> testRuns = testRunDao.searchAll(testRunTemplate);
+            TestDto testTemplate = new TestDto();
+            testTemplate.setTest_suite_id(suiteId);
+            List<TestDto> tests = testController.get(testTemplate);
+            for (TestRunDto testRun : testRuns) {
+                TestResultDto testResultTemplate = new TestResultDto();
+                testResultTemplate.setTest_run_id(testRun.getId());
+                List<TestResultDto> testResults = testResultDao.searchAll(testResultTemplate);
+                tests = tests.stream().filter(test -> {
+                    TestResultDto currentResult = testResults.stream().filter(result -> result.getTest_id().equals(test.getId())).findFirst().orElse(null);
+                    return currentResult == null || currentResult.getFinal_result_id() == 3;
+                }).collect(Collectors.toList());
+            }
+
+            return tests;
+        } else {
+            throw new AqualityPermissionsException("Account is not allowed to Sync Test Suite", baseUser);
```

Comment:
Let's inform user what role necessary to use the feature

Reply:
Don't want do this cos user will never see it if he is not trying to brake the system without permissions